gpytorch/models/exact_gp.py

```diff
@@ -147,33 +148,44 @@ def __call__(self, *args, **kwargs):
                     # non-batch mode (standard)
                     num_train = train_targets.size(-1)
 
-            predictive_mean, mean_cache = exact_predictive_mean(
-                full_covar=full_covar,
-                full_mean=full_mean,
-                train_inputs=train_inputs,
-                train_labels=train_targets,
-                num_train=num_train,
-                likelihood=self.likelihood,
-                precomputed_cache=self.mean_cache,
-                non_batch_train=non_batch_train,
-            )
-            predictive_covar, covar_cache = exact_predictive_covar(
-                full_covar=full_covar,
-                train_inputs=train_inputs,
-                num_train=num_train,
-                likelihood=self.likelihood,
-                precomputed_cache=self.covar_cache,
-                non_batch_train=non_batch_train,
-            )
-
-            self.mean_cache = mean_cache
-            self.covar_cache = covar_cache
-            if num_tasks > 1:
-                if train_targets.ndimension() == 2:
-                    # Batch multitask
-                    predictive_mean = predictive_mean.view(train_targets.size(0), -1, num_tasks).contiguous()
-                else:
-                    # Standard multitask
-                    predictive_mean = predictive_mean.view(-1, num_tasks).contiguous()
+                train_train_covar = full_covar[..., :num_train, :num_train].evaluate_kernel()
+                test_train_covar = full_covar[..., num_train:, :num_train]
+                test_test_covar = full_covar[..., num_train:, num_train:]
+
+                pred_strat = prediction_strategy(train_train_covar)
```

Comment:
Rather than storing the mean_cache and covar_cache, it might make sense to cache pred_strat? This might be cleaner, since then pred_strat can make however many caches it wants.

Reply:
Read the last paragraph of my PR description :-)